typings/ws.d.ts

```diff
@@ -1,7 +1,28 @@
 /// <reference types="node" />
 
+/**
+ * Describes the required shape of WebSocket instances.
+ * @interface WebSocketInstance
+ */
+export interface WebSocketInstance {
+  send: Function;
+  onmessage: Function | null;
+  onopen: Function | null;
+  close: Function | null;
+  onclose: Function | null;
+  onerror: Function | null;
```

Comment:
Upon looking at the [WebSocket API doc](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket), send() and close() are methods on the WebSocket class. Then shouldn't `close: Function` like send. Can it be null?

W.r.t types for all the on* properties, can they be `undefined`, since `null` and `undefined` are different in JS/TS.

By typing the on* properties as Function | null we are saying that the properties will **always be present** in the WebSocketInstance. They will either be a `Function` or `null`. Is it possible for those properties to not be present at all? If yes then they should be `Function | null | undefined`.

Reply:
It can't be null, should probably just be Function like send. FWIW I don't much like using the Function type here, but it is suitably generic I suppose.

on* methods are EventTarget-like interfaces that must be supported or Rhea will not work. EventTarget methods like this are either the listener or null, not undefined. See e.g. the types in [dom.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L16788). Also, allowing undefined would have the unfortunate effect of allowing users to pass in a WS implementation without support for these events, which will not work.

Reply:
I'll update `close: Function` just like send.